learn/core_concepts/indexes.md

```diff
@@ -48,55 +37,107 @@ The `uid` is set at [index creation time](/reference/api/indexes.md#create-an-in
 
-[Learn more about document primary key](/learn/core_concepts/primary_key.md#primary-key-2)
+You can set the primary key for an index or let it be inferred by Meilisearch. Read more about [setting the primary key](/learn/core_concepts/primary_key.md#setting-the-primary-key).
 
-## Relevancy rules
+[Learn more about the primary field](/learn/core_concepts/primary_key.md)
 
-Each index applies its own relevancy rules. All indexes are created with the same built-in ranking rules executed in a default order. Once your first document has been added, the index will record how the attributes must be sorted. Their order of importance will be deduced from their order of appearance in the document.
+## Index settings
 
-For example, suppose your first document lists attributes in the following order:
+Index settings can be thought of as a JSON object containing many different options for customizing search behavior.
+
+You can customize the following index settings:
+
+- [Ranking rules](#ranking-rules)
+- [Distinct attribute](#distinct-attribute)
+- [Synonyms](#synonyms)
+- [Filterable attributes](#filterable-attributes)
+- [Sortable attributes](#sortable-attributes)
+- [Stop words](#stop-words)
+- [Displayed and searchable attributes](#displayed-and-searchable-attributes)
+- [Typo tolerance](#typo-tolerance)
+
+To change index settings, use the [update settings endpoint](/reference/api/settings.md#update-settings) or any of the [child routes](/reference/api/settings.md#all-settings).
+
+### Ranking rules
+
+All indexes are created with the same built-in ranking rules executed in default order:
 
 ```
-id, title, overview, release_date
+[
+  "words",
+  "typo",
+  "proximity",
+  "attribute",
+  "sort",
+  "exactness"
+]
 ```
 
-A document containing matches in its `title` field will be considered more relevant than a document only containing matches in its `overview`.
-
-You can alter the order in which ranking rules take effect, or define custom ranking rules to return certain results first.
+The order of these rules matters: the first rule has the most impact, and the last rule has the least. You can alter this order or define custom ranking rules to return certain results first. This can be done using the [update settings endpoint](/reference/api/settings.md#update-settings) or the [update ranking rules endpoint](/reference/api/ranking_rules.md#update-ranking-rules).
 
 [Learn more about ranking rules](/learn/core_concepts/relevancy.md)
 
-## Synonyms
+### Distinct attribute
+
+If your dataset contains multiple similar documents, you may want to return only one on search. Suppose you have numerous black jackets in different sizes in your `costumes` index; setting `costume_id` as the distinct attribute will mean Meiliserch will not return more than one black jacket with the same `costume_id`.
```

Comment:
Changed the example from `costume_id` to `costume_name`. Since `costume_id` contains `id` it can read as a primary key, in which case there wouldn't be two documents w/ the same `costume_id` and making it distinct attribute would be meaningless.

Not that most readers would understand that distinction at this point. It's more for me and my peace of mind 😂 

```suggestion
If your dataset contains multiple similar documents, you may want to return only one on search. Suppose you have numerous black jackets in different sizes in your `costumes` index; setting `costume_name` as the distinct attribute will mean Meiliserch will not return more than one black jacket with the same `costume_name`.
```